src/OnboardingContext.tsx

```diff
@@ -16,9 +17,11 @@ export const OnboardingContext = createContext<
 >(undefined)
 
 export interface OnboardingContextState {
-  onboardingIsComplete: boolean
+  isOnboardingComplete: boolean
   completeOnboarding: () => void
   resetOnboarding: () => void
+  destinationAfterComplete: Stack
```

Comment:
Could this be a prop on the Stack Component itself? Seems like if we are going to be making a distinction in the flows in which the onboarding screens can appear by mounting two separate instances of the screen we dont need to add to the state in the context, but parameterize the component?

I suppose im a little concerned about adding confusion to the notion of `Onboarding`. Is it the act of enabling all the features we need on first visit to the app? or is it the act of moving through a particular set of screens?  I would say its the former not the later and this conflates those two flows in a way that would be hard to parse as a new reader of the codebase.

I might even consider changing what we call the OnboardingStack now to something like `HowTheAppWorksStack` and consider onboarding to be the flow of `Welcome -> How The App Works -> Activation -> the Main App`

Something like this would also allow us to remove the first `Welcome` screen from the currently named `OnboardingReview` flow which seems a bit out of place.

Reply:
Makes sense @johnschoeman, went down the path of parameterizing the `OnboardingStack`. I think it's more clear. Can you take a look? In the process, we ended up refactoring the `OnboardingStack` to map through onboarding screen data instead of rendering various screens e.g. `Introduction` that in turn render an `OnboardingScreen`. This reduced the amount of strange and cumbersome prop-drilling.

Reply:
Thoughts on renaming in a future PR though? I agree but it seems like a lot for this PR.

Reply:
makes sense to me!